### PR TITLE
Upstream PR #6203

### PR DESCRIPTION
**Pull request by beldmit**
More robust way to build the SRL path.

### CODE REVIEW

**Comment by mspncp** [comment]
ping @petrovr.

File: apps/x509.c [hunk 1]
```diff
@@ -916,8 +916,19 @@ static ASN1_INTEGER *x509_load_serial(const char *CAfile,
     BIGNUM *serial = NULL;
 
     if (serialfile == NULL) {
-        const char *p = strchr(CAfile, '.');
-        size_t len = p != NULL ? (size_t)(p - CAfile) : strlen(CAfile);
+        const char *p = strrchr(CAfile, '.');
+        size_t len;
+        /* Check if the dot is in the file name or in one of subdirectories name */
+        if (p != NULL) {
+# ifdef WINDOWS
```
Comment (mspncp) [inline apps/x509.c]: I'm not sure this is really the ultimate solution. I know that path names in VMS are very different, see for example [test/conf_include_test.c#L23](https://github.com/openssl/openssl/blob/master/test/conf_include_test.c#L23). @levitte, any comments?

Comment (beldmit) [reply apps/x509.c]: I did not find a function to deal with paths in OpenSSL source code. I'm aware about VMS but I think it's better than current state.

Comment (mspncp) [reply apps/x509.c]: I didn't find one either. But having raw string handling and \#ifdef's everywhere in the code where paths need to be treated does not feel like the right solution. So either we should implement proper path handling functions somewhere, or just stick with the simple fix #6566 and document that a file extension is required for the `CAfile` (which is normally the case, anyway).

**Comment by petrovr** [comment]
Matthias St. Pierre wrote:
> ping @petrovr.
> 

Not sure. I expect search for last dot after last directory separator.
OpenSSL function ends_with_dirsep() shows possible directory separators 
per platform.  As is visible there is more then one separator. For 
instance \ and / on WIN32.

Regards,
Roumen Petrov

**Pull request closed without merging**